backend/__tests__/posts.js

```diff
@@ -402,4 +402,106 @@ describe('Posts', () => {
         */
         describe('when deleting a parent post', () => {})
     })
+
+    describe('POST /posts/:id/share', () => {
+        describe('when not authenticated', () => {
+            it('should return response code of 400', async () => {
+                const response = await request(server).post(
+                    '/api/posts/1/share'
+                )
+                expect(response.statusCode).toBe(400)
+            })
+        })
+
+        describe('when the post does not exist', () => {
+            it('should return response code of 404', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const response = await request(server)
+                    .post(`/api/posts/999999/share`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(404)
+            })
+        })
+
+        describe('when a valid user is sharing a post that exists', () => {
+            it('should return response code of 201', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const response = await request(server)
+                    .post(`/api/posts/1/share`)
```

Comment:
Id of the post shouldnt be hardcoded because id is automatically generated so it could be different for everyone

Reply:
I got a little lazy and cut corners there. I think I fixed it. There are some hard coded values for the tests involving unliking, but they shouldn't be a problem.